data/src/test/java/io/homeassistant/companion/android/data/integration/IntegrationRepositoryImplSpec.kt

```diff
@@ -109,5 +115,196 @@ object IntegrationRepositoryImplSpec : Spek({
                 }
             }
         }
+
+        describe("location updated") {
+            beforeEachTest {
+                coEvery { authenticationRepository.getUrl() } returns URL("http://example.com")
+                coEvery { localStorage.getString("webhook_id") } returns "FGHIJ"
+            }
+
+            describe("updateLocation cloud url") {
+                val location = UpdateLocation(
+                    "locationName",
+                    arrayOf(45.0, -45.0),
+                    0,
+                    1,
+                    2,
+                    3,
+                    4,
+                    5
+                )
+                val integrationRequest = IntegrationRequest(
+                    "update_location",
+                    UpdateLocationRequest(
+                        location.locationName,
+                        location.gps,
+                        location.gpsAccuracy,
+                        location.battery,
+                        location.speed,
+                        location.altitude,
+                        location.course,
+                        location.verticalAccuracy
+                    )
+                )
+                beforeEachTest {
+                    coEvery { localStorage.getString("cloud_url") } returns "http://best.com/hook/id"
+                    coEvery { localStorage.getString("remote_ui_url") } returns "http://better.com"
+                    coEvery {
+                        integrationService.updateLocation(
+                            any(), // "http://example.com/api/webhook/FGHIJ",
```

Comment:
Could you test this case :
- [ ] let the `cloud_url` fail but let "remote_ui_url" succeed
- [ ] all url is failing